bot/note_command.go

```diff
@@ -51,13 +51,33 @@ func (cc *NoteCommand) Execute(ctx context.Context, update *objects.Update) {
 	if update.Message.Caption != "" {
 		messageText = update.Message.Caption
 	}
-	noteText := strings.Replace(messageText, "/note", "", 1)
-	if noteText == "" {
-		cc.SetUserState(id, "/note")
-		cc.SendMessage("write your note in the next message", id, false, true)
-		return
+	var pageName string
+	var noteText string
+	if strings.Contains(messageText, "--page"){
+		parts := strings.SplitN(messageText, "\"", 3)
+		if len(parts) < 3 { 
+			cc.SendMessage("Make sure you have enclosed the page name in quotes.", id, false, true)
+			return
+		}
+		pageName = parts[1] 
+		noteText = parts[2]
+		if pageName == "" {
+			cc.SendMessage("No page name specified.", id, false, true)
+			return
+		}
+	} else { 
+		noteText := strings.Replace(messageText, "/note", "", 1)
+		if noteText == "" {
+			cc.SetUserState(id, "/note")
+			cc.SendMessage("write your note in the next message", id, false, true)
+			return
+		}
```

Comment:
the `if ... else` conditions can be simplified like this:
```suggestion
	if !strings.Contains(messageText, "--page"){
		noteText := strings.Replace(messageText, "/note", "", 1)
		if noteText == "" {
			cc.SetUserState(id, "/note")
			cc.SendMessage("write your note in the next message", id, false, true)
			return
		}
	}
    parts := strings.SplitN(messageText, "\"", 3)
	if len(parts) < 3 { 
		cc.SendMessage("Make sure you have enclosed the page name in quotes.", id, false, true)
		return
	}
	pageName = parts[1] 
	noteText = parts[2]
	if pageName == "" {
		cc.SendMessage("No page name specified.", id, false, true)
		return
	} 
```